clang/Util.d

```diff
@@ -77,3 +78,35 @@ mixin template CX ()
         return cx !is CType.init;
     }
 }
+
+extern (C) int mkstemps(char*, int);
+extern (C) int close(int);
+
+File namedTempFile(string prefix, string suffix) 
+{
+    import std.random;
+    import std.file;
+    import std.path;
+    import std.stdio;
+    import std.format;
+    import std.exception;
+
+    void randstr (char[] slice)
+    {
+        for (uint i = 0; i < slice.length; ++i)
+            slice[i] = uniform!("[]")('A', 'Z');
+    }
+
+    string name = format("%sXXXXXXXXXXXXXXXX%s\0", prefix, suffix);
+    char[] path = buildPath(tempDir(), name).dup;
+    const size_t termAnd6XSize = 7;
+    randstr(path[$ - name.length + prefix.length .. $ - suffix.length - termAnd6XSize]);
+
+    int fd = mkstemps(path.ptr, cast(int) suffix.length);
+    scope (exit) close(fd);
+
+    if (fd == -1)
+        throw new ErrnoException("Cannot create temporary file.");
```

Comment:
Minor detail but I'm not a big fan of `ErrnoException`. It's an implementation detail that the error occurred in C.